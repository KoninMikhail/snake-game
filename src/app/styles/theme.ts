import {colors, ThemeColors} from './theme_parts/colors';
import {gaps, ThemeGaps} from './theme_parts/gaps';
import {icons, ThemeIcons} from './theme_parts/icons';
import {ThemeTypography, typography} from './theme_parts/typography';
import {animations, ThemeAnimations} from '@styles/theme_parts/animations';

/**
 * @typedef IBaseTheme
 */
export interface IBaseTheme {
    animations: ThemeAnimations;
    colors: ThemeColors;
    gaps: ThemeGaps;
    icons: ThemeIcons;
    typography: ThemeTypography;
}

/**
 * Theme
 */
export const theme: IBaseTheme = {
    colors,
    animations,
    gaps,
    icons,
    typography,
};
