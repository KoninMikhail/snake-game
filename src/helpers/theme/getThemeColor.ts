import { ThemeColors } from '@styles/theme_parts/colors';
import { IBaseTheme } from '@styles/theme';

/**
 * @name getThemeColor
 *
 * Get the color from theme.
 *
 * @param theme - The theme object
 * @param color - The color you want to get from the theme.
 * @returns A function that takes a theme and a color and returns the color from the theme.
 */
const getThemeColor: GetFromThemeFn<IBaseTheme, ThemeColors> = (theme, color) => {
    return theme.colors[color];
};

export default getThemeColor;
