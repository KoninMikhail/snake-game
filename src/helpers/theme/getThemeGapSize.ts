import {IBaseTheme} from '@styles/theme';
import {ThemeGaps} from '@styles/theme_parts/gaps';

/**
 * @name getThemeGapSize
 *
 * Get the gap size from the theme.
 *
 * @param theme - The theme object
 * @param size - The size of the gap you want to get.
 * @returns A function that takes a theme and a size and returns the theme.gaps[size]
 */
const getThemeGapSize: GetFromThemeFn<IBaseTheme, ThemeGaps> = (theme, size) => theme.gaps[size];

export default getThemeGapSize;
