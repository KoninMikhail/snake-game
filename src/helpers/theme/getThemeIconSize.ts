import { IBaseTheme } from '@styles/theme';
import { ThemeIcons } from '@styles/theme_parts/icons';

/**
 * @name getThemeIconSize
 *
 * Get the icon size from the theme.
 *
 * @param theme - The theme object
 * @param size - The size of the icon you want to get.
 * @returns A function that takes a theme and a size and returns the theme.icons[size]
 */
const getThemeIconSize: GetFromThemeFn<IBaseTheme, ThemeIcons> = (theme, size) => {
    return theme.icons[size];
};

export default getThemeIconSize;
