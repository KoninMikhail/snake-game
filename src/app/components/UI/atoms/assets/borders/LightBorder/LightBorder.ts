import { cloneElement } from 'react';
import { useTheme } from 'styled-components';
import getThemeColor from '@/helpers/theme/getThemeColor';
import { IBorderProps } from '@ui/atoms/assets/borders/Border.interface';

/**
 * @name LightBorder
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use for borders
 *
 * @param children - Element for adding border
 */

export const LightBorder = ({ children }: IBorderProps) => {
    const theme = useTheme();
    const outputColor = getThemeColor(theme, 'light');
    return cloneElement(children, {
        style: {
            border: `0.61em solid ${outputColor}`,
        },
    });
};
