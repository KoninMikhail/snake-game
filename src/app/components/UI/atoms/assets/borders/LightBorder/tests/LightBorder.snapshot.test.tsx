import {render} from '@testing-library/react';
import {ThemeProvider} from 'styled-components';
import {IBaseTheme} from '@styles/theme';
import {LightBorder} from '@ui/atoms/assets/borders/LightBorder/LightBorder';

/**
 * Mocks
 */
const fakeTheme = {
    colors: {
        light: 'white',
    },
} as unknown as IBaseTheme;

/**
 * Tests
 */
describe('LightBorder', () => {
    const testClassName = 'testElement';
    const testId = 'testElementID';

    it('return element with valid styles', () => {
        const { container } = render(
            <ThemeProvider theme={fakeTheme}>
                <LightBorder>
                    <div data-testid={testId} className={testClassName}>
                        Test Element
                    </div>
                </LightBorder>
            </ThemeProvider>
        );
        expect(container).toMatchSnapshot();
    });
});
