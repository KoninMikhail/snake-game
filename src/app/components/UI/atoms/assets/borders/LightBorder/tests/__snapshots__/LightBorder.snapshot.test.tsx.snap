// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LightBorder return element with valid styles 1`] = `
<div>
  <div
    class="testElement"
    data-testid="testElementID"
    style="border: 0.61em solid white;"
  >
    Test Element
  </div>
</div>
`;
